### PR TITLE
feat(dapp-console): `RequireLogin` component

### PR DESCRIPTION
**Pull request by tremarkley**
This PR makes the following changes:
- Creates `RequireLogin` wrapper component that requires a user to be authenticated in order to view the children of the component.
- Wraps the Settings page in the `RequireLogin` component

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#349** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/349?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dapp-console/app/components/Auth/RequireLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      } catch (e) {
+        toast({
+          description: 'Sign in failed.',
+          duration: LONG_DURATION,
+        })
+        captureError(e, 'loginUser')
+        logout()
+      }
+    },
+    onError: (privyError) => {
```
Comment (tremarkley) [inline apps/dapp-console/app/components/Auth/RequireLogin.tsx:27]: this callback gets called when there are errors in the privy auth flow or when the user exits out of the login modal.

File: apps/dapp-console/app/components/Auth/RequireLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { useLogin, useLogout, usePrivy } from '@privy-io/react-auth'
+import { apiClient } from '@/app/helpers/apiClient'
+import { toast } from '@eth-optimism/ui-components'
+import { LONG_DURATION } from '@/app/constants/toast'
+import { captureError } from '@/app/helpers/errorReporting'
+import { useEffect, useState } from 'react'
+
+export const RequireLogin = ({ children }: { children: React.ReactNode }) => {
+  const { authenticated } = usePrivy()
```
Comment (nitaliano) [inline apps/dapp-console/app/components/Auth/RequireLogin.tsx:9]: we should also check to make sure `ready` is `true` before we attempt to verify if the user is authenticated yet or not

**Review by nitaliano — APPROVED** [review]

File: apps/dapp-console/app/components/Auth/RequireLogin.tsx (hunk 2 above)
Comment (tremarkley) [reply apps/dapp-console/app/components/Auth/RequireLogin.tsx]: good catch, updated!

**Pull request merged**